src/resources/Search/Search.ts

```diff
@@ -13,9 +13,16 @@ export default class Search extends Ressource {
     // See: https://platform.cloud.coveo.com/docs?api=SearchApi#!/Search/post_rest_search_v2
     // or : https://docs.coveo.com/en/13/cloud-v2-api-reference/search-api#operation/searchUsingGet
     query(restQueryParameters: any) {
-        return this.api.post<any>(
-            this.buildPath(Search.baseUrl, {organizationId: this.api.organizationId}),
-            restQueryParameters
-        );
+        const {viewAllContent, ...bodyParameters} = restQueryParameters;
+
+        const queryParameters = {
+            organizationId: this.api.organizationId,
+            viewAllContent: undefined,
+        };
+        if (typeof viewAllContent !== 'undefined') {
+            queryParameters.viewAllContent = viewAllContent ? 1 : 0;
+        }
+
+        return this.api.post<any>(this.buildPath(Search.baseUrl, queryParameters), bodyParameters);
     }
```

Comment:
Hmm, I really don't like this `any` for the `restQueryParameters`. It's particularly uninformative for the users.

What prevents us to do it properly and list and put a good and descriptive interface? (Yeah, that'd be a breaking change, but I don't consider that a problem here afaik).

Reply:
I agree with you that instead of `any`, there should be a typed interface for `restQueryParameters`.

In this case, I would prefer to keep this PR as is, and then improve the typing of the `restQueryParameters` as a separate user story.

1. `viewAllContent` support is blocking development of another feature.
2. the `query` call accepts lots of parameters and implementing the interface represents a way bigger change than the current PR.
3. as you mentioned, typing `restQueryParameters` would be a breaking change. Any mistake in the interface definition could break a working implementation. This should be handled with care.

Would that work with you?

Reply:
I agree with Luc. While this is indeed a good idea, it expands the scope of this change and would delay a feature that should be completed asap.

Reply:
If SVCC-402 is planned in your next sprint, you got a deal :)

Otherwise, I understand that there's a feature that depends on it, but unchecked tech debt is my concern here as a maintainer, and I would have to put my foot down 😅 

I do not think that it's such a big undertaking, but I'm not the one coding it so, as a compromise, what I would propose is that you move the any to an `any & SomeInterfaceName` with SomeInterfaceName specifying the accepted value for `viewAllContent` only.

i.e. at least type the key that you use directly in the function.